express/src/controllers/clients.ts

```diff
@@ -158,3 +158,120 @@ export const processChangeServiceNameForm: RequestHandler = async (req, res) =>
     req.session.serviceName = newServiceName;
     res.redirect(`/services/${serviceId}/clients`);
 };
+
+export const showProcessChangePublicKeyForm: RequestHandler = async (req, res) => {
+    res.render("service-details/change-public-key.njk", {
+        serviceId: req.params.serviceId,
```

Comment:
```suggestion
        serviceId: req.context.serviceId,
```
Service ID is avaialble on the request context and won't be available on the params since the clients controller is mounted on `/clients/:clientId`